tests/generated.go

```diff
@@ -591,8 +681,36 @@ func parsePairStorageNew(opts []Pair) (pairStorageNew, error) {
 			}
 			result.HasWorkDir = true
 			result.WorkDir = v.Value.(string)
+		// Default pairs
+		case "default_content_type":
+			defaultPairs = true
```

Comment:
Maybe we can simplify the logic via respect to the order of pairs?

We don't need to take `default_storage_pairs` first any more, instead we always take the order.

- If user input `WithDefaultStoragePairs` and `WithDefaultStorageClass`, we will take `WithDefaultStoragePairs` first.
- If user input `WithDefaultStorageClass` and `WithDefaultStoragePairs`, we will take `WithDefaultStorageClass`.

After this change, we don't need to maintain `defaultPairs` or `enableFeatures`, we can set `HasStorageFeatures` directly.


NOTICE: this change requires an update of GSP.

Reply:
Does this mean that we do not consider the case where there is no intersection，like: 
```
oss.WithDefaultStoragePairs("Standard")
// ...
oss.WithDefaultStoragePairs(oss.DefaultStoragePairs{Write: []types.Pair{ps.WithContentType("application/text")}})
```

Reply:
Yes, we can use them concurrently, they will not affect each other.

Reply:
Maybe we need to keep these two variables `defaultPairs` and `enableFeatures` as when there is more than one default pairs(`DefaultStorageClass` and `DefaultContentType`, or `EnableVirtualDir` and `EnableVirtualLink`), we can't set `HasDefault*Pairs` or `Has*Features` to true every time, otherwise we would ignore some default values？

Reply:
`DefaultStorageClass` and `DefaultContentType` will always be appened into `DefaultStorageParis`, I think setting `HasDefaultStorageParis` is the same with `defaultPairs`.